### ISSUE TITLE
1.9.0-rc1 : 'network create' doesn't throw an error if --ipam-driver points to non existent driver

### ISSUE THREAD

**Issue by pyakpyak** [docker/libnetwork#677]
Whatever string value (even non existent driver name) i specify with --ipam-driver, libnetwork always points to the default ipam.

```
vagrant@vagrant-ubuntu-trusty-64:~$ sudo docker network create -d=nf --ipam-driver=ipam222 nf5
32251d5dc67f075dbebd0dff5a699b798758b31a078177ba0b8065baddffec37
```

I'm testing with the following docker version
```
vagrant@vagrant-ubuntu-trusty-64:~$ sudo docker version
Client:
 Version:      1.9.0-rc1
 API version:  1.21
 Go version:   go1.4.2
 Git commit:   9291a0e
 Built:        Wed Oct 14 03:04:22 UTC 2015
 OS/Arch:      linux/amd64

Server:
 Version:      1.9.0-rc1
 API version:  1.21
 Go version:   go1.4.2
 Git commit:   9291a0e
 Built:        Wed Oct 14 03:04:22 UTC 2015
 OS/Arch:      linux/amd64
vagrant@vagrant-ubuntu-trusty-64:~$
```

I'm actually not sure if --ipam-driver parameter works at all as i don't see any attempts to connect to my ipam driver neither in the docker daemon logs (-D) nor in my ipam driver.

**Comment by aboch** [comment]
Thank you for your report !

It looks like when no `--subnet` option is specified, libnetwork `NewNetwork()` is not being passed any `NetworkOptionIpam()` setter.
I am guessing the issue could be in the client code `CmdNetworkCreate()`

We will fix it as part of next libnetwork vendoring in docker.

**Comment by aboch** [comment]
Bug is not in client code, it's in 
`func (daemon *Daemon) CreateNetwork(name, driver string, ipam network.IPAM) (libnetwork.Network, error) `

Ipam option setter is not passed if ip configuration is empty.

```
	if len(ipam.Config) > 0 {
		nwOptions = append(nwOptions, libnetwork.NetworkOptionIpam(ipam.Driver, "", v4Conf, v6Conf))
	}
```

Will fix it as part of next libnetwork vendoring.
Thanks.

**Comment by dave-tucker** [comment]
Thanks @aboch you are a :metal::star:

**Comment by aboch** [comment]
Closed by [docker/docker #17202](https://github.com/docker/docker/pull/17202)
